cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Insert.hs

```diff
@@ -114,8 +117,13 @@ insertShelleyBlock syncEnv shouldLog withinTwoMins withinHalfHour blk details is
 
     let zippedTx = zip [0 ..] (Generic.blkTxs blk)
     let txInserter = insertTx tracer cache (getInsertOptions syncEnv) (getNetwork syncEnv) isMember blkId (sdEpochNo details) (Generic.blkSlotNo blk)
-    grouped <- foldM (\grouped (idx, tx) -> txInserter idx tx grouped) mempty zippedTx
-    minIds <- insertBlockGroupedData tracer grouped
+    (blockGroupedData, txFees) <- foldM (\gp (idx, tx) -> txInserter idx tx gp) (mempty, 0) zippedTx
+    minIds <- insertBlockGroupedData tracer blockGroupedData
+
+    -- now that we've inserted all the txs for a ShellyBlock lets put what we need in the cache
+    latestEpochFromDb <- lift queryLatestEpoch
```

Comment:
yeah I was thinking the same. Technically it's there once the epoch has been put into the db, put I put this here as a fail safe. Maybe I should abstract this out as a function and it can call the db when it isn't in cache.